File: src/main/java/com/sliusar/todolist/models/Schedule.java
package com.sliusar.todolist.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Setter
@Getter
@EqualsAndHashCode
@ToString(exclude = "todoList")
public class Schedule {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String title;
    private Date createdAt;
    private Date updatedAt;

    @OneToMany(fetch = FetchType.LAZY, cascade = {CascadeType.ALL})
    @JoinColumn(name = "schedule_id")
    private List<Todo> todoList = new ArrayList<>();
}

File: src/main/frontend/src/app/models/schedule.ts
import {Todo} from "./todo";

export class Schedule {
  id: number | undefined;
  title: string;
  createdAt: Date;
  updatedAt: Date;
  todoList: Todo[];

  constructor(title: string, createdAt: Date, updatedAt: Date, todoList: Todo[]) {
    this.title = title;
    this.createdAt = createdAt;
    this.updatedAt = updatedAt;
    this.todoList = todoList;
  }
}
